diff --git a/./src/angular/_jade/ts/latest/guide/index.jade b/./src/angular/guide/index.jade
index b9c74f9..89bec5a 100644
--- a/./src/angular/_jade/ts/latest/guide/index.jade
+++ b/./src/angular/guide/index.jade
@@ -1,19 +1,21 @@
-block includes
-  //- include ../_util-fns
-  - var _angular_io = 'angular.io';
-
-- var __lang = _docsFor || current.path[1] || 'ts';
-- var guideData = {}; // public.docs[__lang].latest.guide._data;
-- var advancedLandingPage = 'attribute-directives'; // '';
-- for(var page in guideData) {
--   if (!guideData[page].basics && !guideData[page].hide) { advancedLandingPage = page; break; }
-- }
-- var advancedUrl = './' + advancedLandingPage + '.html'
-
-
+---
+layout: angular
+title: Documentation Overview
+description: How to read and use this documentation
+sideNavGroup: basic
+prevpage:
+  title: About AngularDart
+  url: /angular
+nextpage:
+  title: Setup for Development
+  url: /angular/guide/setup
+---
+//- FilePath: src/angular/guide/index.jade
+include ../_util-fns
+- var advancedUrl = './attribute-directives'
 :marked
   This page describes the Angular documentation at a high level.
-  If you're new to Angular, you may want to visit "[Learning Angular](learning-angular.html)" first.
+  If you're new to Angular, you may want to visit [Learning Angular](learning-angular) first.
 
   ## Themes
 
@@ -27,25 +29,17 @@ table(width="100%")
   col(width="15%")
   col
   tr(style=top)
-    td <b><a href="../quickstart.html">QuickStart</a></b>
-    td
-      :marked
-        A first taste of Angular<span if-docs="ts"> with zero installation. 
-        Run "Hello World" in an online code editor and start playing with live code</span>.
-  tr(style=top)
     td <b>Guide</b>
     td
       :marked
-        Learn the Angular basics (you're already here!) like the setup for local development,
-        displaying data and accepting user input, injecting application services into components,
-        and building simple forms.
+        Learn the Angular basics (you're already here!) like the
+        [setup](./setup) for local development,
+        [displaying data](./displaying-data) and
+        accepting [user input](./user-input),
+        [injecting application services](./dependency-injection) into components,
+        and building simple [forms](./forms).
   tr(style=top)
-    td <b><a href="../api/">API Reference</a></b>
-    td
-      :marked
-        Authoritative details about each of the Angular libraries.
-  tr(style=top)
-    td <b><a href="../tutorial/">Tutorial</a></b>
+    td <b><a href="../tutorial">Tutorial</a></b>
     td
       :marked
         A step-by-step, immersive approach to learning Angular that
@@ -55,11 +49,11 @@ table(width="100%")
     td
       :marked
         In-depth analysis of Angular features and development practices.
-  tr(style=top if-docs="ts")
-    td <b><a href="../cookbook/">Cookbook</a></b>
+  tr(style=top)
+    td <b><a href="../api">API Reference</a></b>
     td
       :marked
-        Recipes for specific application challenges, mostly code snippets with a minimum of exposition.
+        Authoritative details about each of the Angular libraries.
 
 :marked
   A few early pages are written as tutorials and are clearly marked as such.
@@ -77,19 +71,17 @@ table(width="100%")
   The link launches a browser-based, code editor where you can inspect, modify, save, and download the code.
   </span>
 
-  Alternatively, you can run the example locally, next to those `live-example` links you have a <a href="/resources/zips/architecture/architecture.zip">download link</a>.
-  Just download, unzip, run `npm install` to install the dependencies and run it with `npm start`.
-
   ## Reference pages
 
-  * The [Cheat Sheet](cheatsheet.html) lists Angular syntax for common scenarios.
-  * The [Glossary](glossary.html) defines terms that Angular developers should know.
-  <li if-docs="ts">The [Change Log](change-log.html) announces what's new and changed in the documentation.</li>
-  * The [API Reference](../api/) is the authority on every public-facing member of the Angular libraries.
+  * The [Glossary](glossary) defines terms that Angular developers should know.
+  * The [Cheat Sheet](cheatsheet) lists Angular syntax for common scenarios.
+  * The [API Reference](../api) is the authority on every public-facing member of the Angular libraries.
 
   ## Feedback
 
-  We welcome feedback! 
+  Please tell us about any issues you find:
 
-  * Use the <a href="!{_ngDocRepoURL}" target="_blank" rel="noopener" title="angular docs on github">!{_angular_io} Github repository</a> for **documentation** issues and pull requests.
-  * Use the <a href="!{_ngRepoURL}" target="_blank" rel="noopener" title="angular source on github">Angular Github repository</a> to report issues with **Angular** itself.
+  * For **documentation and example** issues, use the
+    [site-webdev issue tracker.](!{_ngDocRepoURL}/issues)
+  * To report issues with **AngularDart** itself, use the
+    [angular2 issue tracker.](!{_ngRepoURL}/issues)
