diff --git a/./src/angular/_jade/ts/latest/guide/pipes.jade b/./src/angular/guide/pipes.jade
index 95ae371..f12eabe 100644
--- a/./src/angular/_jade/ts/latest/guide/pipes.jade
+++ b/./src/angular/guide/pipes.jade
@@ -1,5 +1,19 @@
-block includes
-  //- include ../_util-fns
+---
+layout: angular
+title: Pipes
+description: Pipes transform displayed values within a template.
+sideNavGroup: advanced
+prevpage:
+  title: Lifecycle Hooks
+  url: /angular/guide/lifecycle-hooks
+nextpage:
+  title: Router
+  url: /angular/guide/router
+---
+//- FilePath: src/angular/guide/pipes.jade
+
+- var _example = 'pipes';
+include ../_util-fns
 
 :marked
   Every application starts out with what seems like a simple task: get data, transform them, and show them to users.
@@ -17,8 +31,7 @@ block includes
   In fact, you might like to apply them in your HTML templates as you do styles.
 
   Introducing Angular pipes, a way to write display-value transformations that you can declare in your HTML.
-  
-  You can run the <live-example></live-example> in Plunker and download the code from there.
+  Try the <live-example></live-example>.
 
 .l-main-section
 :marked
@@ -144,7 +157,7 @@ figure.image-display
   * To tell Angular that this is a pipe, you apply the
   `@Pipe` #{_decorator}, which you import from the core Angular library.
   * The `@Pipe` #{_decorator} allows you to define the
-     pipe name that you'll use within template expressions. It must be a valid JavaScript identifier.
+     pipe name that you'll use within template expressions. It must be a valid Dart identifier.
      Your pipe's name is `exponentialStrength`.
 
 .l-sub-section
@@ -285,8 +298,8 @@ figure.image-display
   ### Pure pipes
 
   Angular executes a *pure pipe* only when it detects a *pure change* to the input value.
-  A pure change is either a change to a primitive input value (`String`, `Number`, `Boolean`, `Symbol`)
-  or a changed object reference (`Date`, `Array`, `Function`, `Object`).
+  In AngularDart, a *pure change* results only from a change in object reference 
+  (given that [everything is an object in Dart](https://www.dartlang.org/docs/dart-up-and-running/ch02.html#important-concepts)).
 
   Angular ignores changes within (composite) objects.
   It won't call a pure pipe if you change an input month, add to an input #{_array}, or update an input object property.
