diff --git a/./src/angular/_jade/ts/latest/guide/security.jade b/./src/angular/_jade/ts/_cache/guide/security.jade
index ee65070..be2c66b 100644
--- a/./src/angular/_jade/ts/latest/guide/security.jade
+++ b/./src/angular/_jade/ts/_cache/guide/security.jade
@@ -21,7 +21,7 @@ block includes
     * [Auditing Angular applications](#code-review).
 
 :marked
-  You can run the <live-example></live-example> in Plunker and download the code from there.
+  Try the <live-example></live-example> of the code shown in this page.
 
 .l-main-section
 h2#report-issues Reporting vulnerabilities
@@ -37,14 +37,14 @@ h2#best-practices Best practices
   * **Keep current with the latest Angular library releases.**
   We regularly update the Angular libraries, and these updates may fix security defects discovered in
   previous versions. Check the Angular [change
-  log](https://github.com/angular/angular/blob/master/CHANGELOG.md) for security-related updates.
+  log](https://github.com/dart-lang/angular2/blob/master/CHANGELOG.md) for security-related updates.
 
   * **Don't modify your copy of Angular.**
   Private, customized versions of Angular tend to fall behind the current version and may not include
   important security fixes and enhancements. Instead, share your Angular improvements with the
   community and make a pull request.
 
-  * **Avoid Angular APIs marked in the documentation as “_Security Risk_.”**
+  * **Avoid Angular APIs marked in the documentation as possible “_security risks_.”**
   For more information, see the [Trusting safe values](#bypass-security-apis) section of this page.
 
 .l-main-section
@@ -140,8 +140,6 @@ block html-sanitization
   and greatly improves application performance. Use the offline template compiler in production
   deployments; don't dynamically generate templates. Angular trusts template code, so generating
   templates, in particular templates containing user data, circumvents Angular's built-in protections. 
-  For information about dynamically constructing forms in a safe way, see the 
-  [Dynamic Forms](../cookbook/dynamic-form.html) cookbook page.
 
   ### Server-side XSS protection
 
@@ -163,7 +161,7 @@ block bypass-security-apis
     are introducing a security vulnerability into your application. If in doubt, find a professional 
     security reviewer.
 
-    To mark a value as trusted, inject `DomSanitizer` and call one of the
+    To mark a value as trusted, inject `DomSanitizationService` and call one of the
     following methods:
 
     * `bypassSecurityTrustHtml`
