diff --git a/./src/angular/_jade/ts/latest/guide/structural-directives.jade b/./src/angular/guide/structural-directives.jade
index 26cf869..a9ffb9f 100644
--- a/./src/angular/_jade/ts/latest/guide/structural-directives.jade
+++ b/./src/angular/guide/structural-directives.jade
@@ -1,30 +1,21 @@
-block includes
-  //- include ../_util-fns
+---
+layout: angular
+title: Structural Directives
+description: Angular has a powerful template engine that lets us easily manipulate the DOM structure of our elements.
+sideNavGroup: advanced
+prevpage:
+  title: Security
+  url: /angular/guide/security
+---
+//- FilePath: src/angular/guide/structural-directives.jade
 
-//- The docs standard h4 style uppercases, making code terms unreadable. Override it.
-style.
-  h4 {font-size: 17px !important; text-transform: none !important;}
-  .syntax { font-family: Consolas, 'Lucida Sans', Courier, sans-serif; color: black; font-size: 85%; }
+- var _example = 'structural-directives';
+include ../_util-fns
 
 :marked
   This guide looks at how Angular manipulates the DOM with **structural directives** and 
   how you can write your own structural directives to do the same thing.
 
-  ### Table of contents
-
-  - [What are structural directives?](#definition)
-  - [*NgIf* case study](#ngIf)
-  - [The asterisk (*) prefix](#asterisk)
-  - [Inside *NgFor*](#ngFor)
-    - [microsyntax](#microsyntax)
-    - [template input variables](#template-input-variable)
-    - [one structural directive per element](#one-per-element)
-  - [Inside the *NgSwitch* directives](#ngSwitch)
-  - [Prefer the (*) prefix](#prefer-asterisk)
-  - [The &lt;template> element](#template)
-  - [Group sibling elements with &lt;ng-container&gt;](#ng-container)
-  - [Write a structural directive](#unless)
-
   Try the <live-example></live-example>.
 
 a#definition
@@ -381,7 +372,7 @@ a#ngcontainer
 a#ng-container
 .l-main-section
 :marked
-  ## Group sibling elements with &lt;ng-container&gt;
+  ## Group sibling elements
 
   There's often a _root_ element that can and should host the structural directive.
   The list element (`<li>`) is a typical host element of an `NgFor` repeater.
@@ -465,9 +456,9 @@ figure.image-display
 :marked
   The `<ng-container>` is a syntax element recognized by the Angular parser.
   It's not a directive, component, class, or interface. 
-  It's more like the curly braces in a JavaScript `if`-block:
+  It's more like the curly braces in a Dart `if`-block:
 
-code-example(language="javascript").
+code-example(language="dart").
   if (someCondition) {
     statement1;
     statement2;
@@ -475,7 +466,7 @@ code-example(language="javascript").
   } 
 
 :marked
-  Without those braces, JavaScript would only execute the first statement
+  Without those braces, Dart would only execute the first statement
   when you intend to conditionally execute all of them as a single block.
   The `<ng-container>` satisfies a similar need in Angular templates.
 
@@ -493,15 +484,6 @@ a#unless
 
 :marked
   Creating a directive is similar to creating a component.
-
-  * Import the `Directive` decorator (instead of the `Component` decorator).
-
-  * Import the `Input`, `TemplateRef`, and `ViewContainerRef` symbols; you'll need them for _any_ structural directive.
-
-  * Apply the decorator to the directive class.
-
-  * Set the CSS *attribute selector* that identifies the directive when applied to an element in a template.
-
   Here's how you might begin:
 
 +makeExcerpt('app/unless.directive.ts (skeleton)')
@@ -517,7 +499,7 @@ a#unless
   In this example, the prefix is `my`.
 
 :marked
-  The directive _class_ name ends in `Directive` per the [style guide](style-guide.html#02-03 "Angular Style Guide").
+  The directive _class_ name ends in `Directive`.
   Angular's own directives do not.
 
   ### _TemplateRef_ and _ViewContainerRef_
@@ -592,7 +574,6 @@ a#summary
   structural-directives/ts/app/app.component.ts,
   structural-directives/ts/app/app.component.html,
   structural-directives/ts/app/app.component.css,
-  structural-directives/ts/app/app.module.ts,
   structural-directives/ts/app/hero.ts,
   structural-directives/ts/app/hero-switch.components.ts,
   structural-directives/ts/app/unless.directive.ts
@@ -601,7 +582,6 @@ a#summary
   `app.component.ts,
    app.component.html,
    app.component.css,
-   app.module.ts, 
    hero.ts,
    hero-switch.components.ts,
    unless.directive.ts
